### ISSUE TITLE
Wifi Disc Streaming not working on Ubuntu

### ISSUE THREAD

**Issue by hrydgard** [hrydgard/ppsspp#11853]
Tried to stream ISOs from my Windows machine to a Linux box, and it fails with errno 32 (broken pipe). Same wifi. Will investigate more later, just writing this down as a reminder.

**Comment by unknownbrackets** [comment]
Maybe need to ignore sigpipe in ext/native/net/sinks.cpp?  #11475 - or just ignore the signal entirely / use SO_NOSIGPIPE.  That said, I wonder why the connection is being closed early...

-[Unknown]

**Comment by hrydgard** [comment]
Yeah. Additionally, at the very least we should be able to report a better error, when it is a real one.

**Comment by hrydgard** [comment]
MSG_NOSIGNAL doesn't help of course - we never got a signal, just an error code from send. Still, a bit mysterious. maybe some firewall nonsense

**Comment by unknownbrackets** [comment]
If you're able to access from a web browser on Ubuntu (even if it just gives the listing, it won't let you download the actual iso), it should be working and rule out any firewall shenanigans.  If not, that might give us a better error message we could use to improve the error message PPSSPP gives...

-[Unknown]

**Comment by unknownbrackets** [comment]
Probably a firewall.  It's working fine on Raspbian (both ways.)

Windows will firewall things depending on how you connected to the network (i.e. if you selected 'Home' or not), and what access you gave PPSSPP when it first tried to listen on the network.

-[Unknown]

**Comment by hrydgard** [comment]
Yup, turned out to be. However I'm leaving open with the intention to improve error messaging.

**Comment by unknownbrackets** [comment]
I think the Windows Firewall detection has greatly improved this for most users.  Going to bump further stuff to the next version.

-[Unknown]
